keras_cv/layers/fpn/feature_pyramid.py

```diff
@@ -0,0 +1,156 @@
+# limitations under the License.
+
+import tensorflow as tf
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class FeaturePyramid(tf.keras.layers.Layer):
+    """Implements a Feature Pyramid Network.
+
+    Feature Pyramid Networks (FPNs) are basic components that are added to an
+    existing CNN to extract features at different scales. For the basic FPN, the
+    inputs are features `Ci` from different levels of a CNN, where the feature is
+    scaled from the image size by `1/2^i` for any level `i`.
+
+    There is an output associated with each level in the basic FPN. The output Pi
+    at level `i` (corresponding to Ci) is given by performing a merge operation on
+    the outputs of:
+    1) a top down operation on Pj (where j = i - 1)
+    2) a lateral operation on Ci
+
+    The layer must be called with the feature Ci in increasing order of `i`. As an
+    example, for `pyramid_levels`=[2,3,4,5], the layer should be called with the
+    list [C2,C3,C4,C5] as input
+
+    Args:
+        pyramid_levels: a python list that specifies all the values of level `i`
+            for which the feature Ci will be specified. The size of this list is
+            num_pyramid_levels.
+        num_channels: an integer representing the number of channels for the FPN
+            operations. Defaults to 256.
+        lateral_ops: a python list of size num_pyramid_levels that specifies all the
+            lateral operations performed by the FPN. Each operation from the list must
+            match the corresponding level from pyramid_levels list. Passing None will
+            populate the list with the default FPN operation (a 1X1 Conv2D layer).
+            Defaults to None.
+        top_down_ops: a python list of size num_pyramid_levels that specifies all the
+            top down operations performed by the FPN. Each operation from the list must
+            match the corresponding level from pyramid_levels list. Passing None will
+            populate the list with the default FPN operation (an UpSampling2D layer).
+            Defaults to None.
+        merge_ops: a python list of size (num_pyramid_levels - 1) that specifies all
+            the merge operations performed by the FPN. Each operation from the list must
+            match the corresponding level from pyramid_levels list. The top-most layer
+            is not merged and the corresponding operation is therefore not included.
+            Passing None will populate the list with the default FPN operation
+            (an add operation). Defaults to None.
+
+    References:
+        [FPN paper](https://arxiv.org/pdf/1612.03144)
+
+    Sample Usage:
+    ```python
+    inp = tf.keras.layers.Input((384, 384, 3))
+    efnb0 = tf.keras.applications.EfficientNetB0(input_tensor=inp, include_top = False)
+
+    layer_names = ['block2b_add', 'block3b_add', 'block5c_add', 'top_activation']
+    backbone_outputs = [efnb0.get_layer(name).output for name in layer_names]
```

Comment:
Thanks for this example.

@tanzhenyu, I think we should have some util/function to extract common feature layers from model backbones. (those layer names should be accessible from the model itself.)